lib/ansible/module_utils/ismount.py

```diff
@@ -49,42 +49,71 @@
 # Agreement.
 
 import os
+import sys
 
 
 def ismount(path):
-    """Test whether a path is a mount point
-    clone of os.path.ismount (from cpython Lib/posixpath.py)
-    fixed to solve https://github.com/ansible/ansible-modules-core/issues/2186
-    and workaround non-fixed http://bugs.python.org/issue2466
-    this should be rewritten as soon as python issue 2466 is fixed
-    probably check for python version and use os.path.ismount if fixed
+    """python issue 2466 is fixed then ismount() was rewritten"""
+    # check python version
 
-    to remove replace in this file ismount( -> os.path.ismount( and remove this
-    function"""
+    current_version = sys.version_info
+    if (current_version.major == 3 and 5 <= current_version.minor <= 6) or (current_version.major == 2 and current_version.minor == 7):
```

Comment:
After thinking about this more, it's probably more reliable and easier to maintain if we just default to always using our version of `ismount()` rather that trying to evaluate the Python version.

Ideally we could probe somehow and fallback to this version rather than doing a version comparison, but it's really hard to probe for a bugfix.
```suggestion

```